sycl/source/detail/kernel_data.hpp

```diff
@@ -163,6 +172,131 @@ class KernelData {
     MKernelWorkGroupMemorySize = Size;
   }
 
+  void parseAndSetCacheConfigProperty(
+      const sycl::ext::intel::experimental::cache_config_key &prop) {
+    using namespace sycl::ext::intel::experimental;
+
+    ur_kernel_cache_config_t CacheConfig =
+        ur_kernel_cache_config_t::UR_KERNEL_CACHE_CONFIG_DEFAULT;
+    if (prop == large_slm) {
+      CacheConfig = ur_kernel_cache_config_t::UR_KERNEL_CACHE_CONFIG_LARGE_SLM;
+    } else if (prop == large_data) {
+      CacheConfig = ur_kernel_cache_config_t::UR_KERNEL_CACHE_CONFIG_LARGE_DATA;
+    } else
+      assert(false && "unknown cache property type");
+
+    MKernelCacheConfig = CacheConfig;
+  }
+
+  template <int ClusterDims>
+  void parseAndSetClusterDimProperty(
+      const std::optional<cluster_size_key<ClusterDims>> &prop) {
+    if (prop) {
+      static_assert(ClusterDims < 4 && ClusterDims > 0,
+                    "Invalid cluster dimensions");
+
+      auto ClusterSize = prop->get_cluster_size();
+      MKernelUsesClusterLaunch = true;
+
+      if constexpr (ClusterDims == 1)
+        MNDRDesc.setClusterDimensions(sycl::range<1>{ClusterSize[0]});
+      else if constexpr (ClusterDims == 2)
+        MNDRDesc.setClusterDimensions(
+            sycl::range<2>{ClusterSize[0], ClusterSize[1]});
+      else if constexpr (ClusterDims == 3)
+        MNDRDesc.setClusterDimensions(
+            sycl::range<3>{ClusterSize[0], ClusterSize[1], ClusterSize[2]});
+    }
+  }
+
+  void validateAndSetKernelLaunchProperties(
+      const detail::KernelPropertyHolderStructTy &Kprop, bool HasGraph,
+      const device_impl &dev) {
```

Comment:
Are you sure we still need this? Can we pass `Kprop` as-is down to the UR calls?